### PR TITLE
Adapt to new project config design

### PR DESCRIPTION
**Pull request by timja**
Requires https://github.com/jenkinsci/jenkins/pull/6485

### CODE REVIEW

**Comment by janfaracik** [comment]
Heya, I'm also doing some work atm for this, mind if I merge mine into yours/could always do a second PR if you prefer?

**Comment by timja** [comment]
> Heya, I'm also doing some work atm for this, mind if I merge mine into yours/could always do a second PR if you prefer?

fine to push directly, I've sent you an invite

File: src/main/resources/com/cloudbees/hudson/plugins/folder/AbstractFolder/configure.jelly [hunk 1]
```diff
@@ -22,48 +22,72 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
...
+            <h2>
+              <div class="jenkins-hidden">
+                <l:icon src="symbol-settings" />
+              </div>
+              <span>${%General}</span>
+            </h2>
+          </div>
+          <j:if test="${it.supportsMakeDisabled()}">
+            <div class="jenkins-app-bar__controls">
+              <p:config-disableBuild tooltip="${%blurb(it.pronoun)}" />
```
Comment (timja) [inline src/main/resources/com/cloudbees/hudson/plugins/folder/AbstractFolder/configure.jelly:62]: was there a reason you were using `toggleSwitch` directly @janfaracik ? this seemed to support tooltip already just fine

File: pom.xml [hunk 2]
```diff
@@ -22,7 +22,7 @@
     <changelist>999999-SNAPSHOT</changelist>
     <gitHubRepo>jenkinsci/${project.artifactId}-plugin</gitHubRepo>
     <!-- remember to change the io.jenkins.tools.bom artifact when changing this -->
-    <jenkins.version>2.277.1</jenkins.version>
+    <jenkins.version>2.348-SNAPSHOT</jenkins.version> <!-- TODO https://github.com/jenkinsci/jenkins/pull/6485 -->
```
Comment (timja) [inline pom.xml:25]: no incremental produced at all recently that I can tell

File: src/main/resources/com/cloudbees/hudson/plugins/folder/AbstractFolder/configure.jelly (hunk 1 above)
Comment (janfaracik) [reply src/main/resources/com/cloudbees/hudson/plugins/folder/AbstractFolder/configure.jelly:62]: Ah think it was just to change the `name` property, this is a better solution though!

Comment (timja) [reply src/main/resources/com/cloudbees/hudson/plugins/folder/AbstractFolder/configure.jelly:62]: name matches core 🤷

**Review by alecharp — APPROVED** [review]
This is looking great. Tried all the "default" configuration on folders and didn't see any regression.This is looking great. Tried all the "default" configuration on folders and didn't see any regression.

**Review by NotMyFault — APPROVED** [review]

**Comment by alecharp** [comment]
@timja I updated the `jenkins.version` to `2.359` to include the configure project redesign.

File: pom.xml [hunk 3]
```diff
@@ -22,7 +22,7 @@
     <changelist>999999-SNAPSHOT</changelist>
     <gitHubRepo>jenkinsci/${project.artifactId}-plugin</gitHubRepo>
     <!-- remember to change the io.jenkins.tools.bom artifact when changing this -->
-    <jenkins.version>2.319.1</jenkins.version>
+    <jenkins.version>2.359</jenkins.version>
```
Comment (basil) [inline pom.xml:25]: Needs corresponding changes to the `Jenkinsfile` as in https://github.com/jenkinsci/sshd-plugin/blob/2dc2ea416e3300ad210e3943b471204d562e6d18/Jenkinsfile

Comment (jglick) [reply pom.xml:25]: (You would need write access to test that. If you do not have write access, I can refile as a separate testing-only PR. Ideally `github-branch-source` could be instructed by a maintainer somehow to test a given commit.)

Comment (timja) [reply pom.xml:25]: conveniently I have global admin 😄.

and yes that would be good

**Review by basil — APPROVED** [review]

**Comment by alecharp** [comment]
I tested the plugin locally, custom icons, moving projects and folders, and I couldn't find any problem. 
We can merge this. Any objections?

File: src/main/resources/com/cloudbees/hudson/plugins/folder/AbstractFolder/tasks-top.jelly [hunk 4]
```diff
@@ -25,7 +25,8 @@ THE SOFTWARE.
 <j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout"
          xmlns:t="/lib/hudson" xmlns:f="/lib/form" xmlns:i="jelly:fmt" xmlns:this="this">
   <j:set var="url" value="${h.getNearestAncestorUrl(request,it)}"/>
-  <l:task icon="icon-search icon-md"  href="${url}/" title="${%Status}" contextMenu="false" />
+  <l:task icon="icon-up icon-md" href="${url}/../../" title="${%Up}" contextMenu="false"/>
```
Comment (jglick) [inline src/main/resources/com/cloudbees/hudson/plugins/folder/AbstractFolder/tasks-top.jelly:28]: I thought we were getting rid of these? Or is that a separate PR?

**Review by jglick — APPROVED** [review]
Deps look fine to me. @alecharp feel free to merge (and thus release).

File: src/main/resources/com/cloudbees/hudson/plugins/folder/AbstractFolder/tasks-top.jelly (hunk 4 above)
Comment (timja) [reply src/main/resources/com/cloudbees/hudson/plugins/folder/AbstractFolder/tasks-top.jelly:28]: separate, https://github.com/jenkinsci/cloudbees-folder-plugin/pull/252 I suspect a bad merge resolution

**Comment by alecharp** [comment]
Thanks @timja !

**Comment by jglick** [comment]
> I suspect a bad merge resolution

Tip: add

```ini
[merge]
	conflictstyle = diff3
```

to `~/.gitconfig`. (Why this is not the default, I cannot fathom.)

**Comment by alecharp** [comment]
This was on me. I did the merge in GitHub directly (which seems to be even worse)

**Comment by jglick** [comment]
Yeah IIRC it shows the diff2 conflict markers only, which is basically useless—like trying to answer an analogy question

> _…some stuff you can look up if you really care…_ is to cat as hoof is to what?

**Comment by basil** [comment]
Great point. I used to ask people to resolve [a merge conflict like this](https://psung.blogspot.com/2011/02/reducing-merge-headaches-git-meets.html) during interviews (with diff3 enabled, of course!) as a test of analytical reasoning skills.

**Pull request merged**